java/src/main/java/com/genexus/internet/GXRestAPIClient.java

```diff
@@ -506,6 +506,9 @@ public void RestExecute() {
 				httpClient.addHeader("Content-Type", contentType);
 			}
 		}
+		if (this.location.getAuthenticationMethod() == 4 && this.location.getAccessToken() != null &&  ! this.location.getAccessToken().trim().isEmpty())  {
```

Comment:
Why do you use this.location instead of only location?
The same with this.httpMethod